lib/container/Circular_Buffer.fz

```diff
@@ -0,0 +1,96 @@
+# This file is part of the Fuzion language implementation.
+#
+# The Fuzion language implementation is free software: you can redistribute it
+# and/or modify it under the terms of the GNU General Public License as published
+# by the Free Software Foundation, version 3 of the License.
+#
+# The Fuzion language implementation is distributed in the hope that it will be
+# useful, but WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public
+# License for more details.
+#
+# You should have received a copy of the GNU General Public License along with The
+# Fuzion language implementation.  If not, see <https://www.gnu.org/licenses/>.
+
+
+# -----------------------------------------------------------------------
+#
+#  Tokiwa Software GmbH, Germany
+#
+#  Source code of Fuzion standard library feature Circular_Buffer
+#
+# -----------------------------------------------------------------------
+
+# Circular_Buffer -- a circular buffer-like structure
+#
+module:public Circular_Buffer
+  (# type of elements stored in this buffer
+   T type,
+
+   # effect used to modify this buffer
+   E type : effect) ref
+
+is
+
+
+  # length of this buffer.
+  #
+  public length i64 => abstract
+
+
+  # amount of elements free for writing
+  #
+  public available i64 =>
+    abstract
+
+
+  # amount of elements available for reading
+  public buffered i64 =>
+    abstract
+
+
+  # enqueue a single element into the buffer
+  #
+  public put (e T) outcome unit =>
+    abstract
+
+
+  # read a single element from the buffer
+  #
+  public get outcome T =>
```

Comment:
I wonder if `option` instead of `outcome` is the better choice here? I mean what could you do with a specific error? I can't think of something?